src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.iOS.cs

```diff
@@ -258,6 +259,14 @@ private UIStringAttributes GetAttributes()
 				attributes.KerningAdjustment = (CharacterSpacing / 1000f) * 12;
 			}
 
+			// Foreground checks should be kept at the end since we **may** use the attributes to calculate text size
+			// for gradient brushes.
+			if (Foreground is SolidColorBrush solidColorBrush)
+			{
+				attributes.ForegroundColor = solidColorBrush.ColorWithOpacity;
+			}
+			// TODO: Support other brushes.
+
```

Comment:
Instead, you should use `Brush.TryGetColorWithOpacity(ForeGround, out var color)`;